coordinator/src/main/scala/filodb.coordinator/queryengine2/QueryEngine.scala

```diff
@@ -198,10 +198,18 @@ class QueryEngine(dataset: Dataset,
                                       submitTime: Long,
                                       options: QueryOptions,
                                       lp: LabelValues): Seq[LabelValuesExec] = {
-    shardsFromFilters(lp.filters, options).map { shard =>
+    val filters = lp.labelConstraints.map { case (k, v) =>
+      new ColumnFilter(k, Filter.Equals(v))
+    }.toSeq
+    val shardsToHit = if (shardColumns.toSet.subsetOf(lp.labelConstraints.keySet)) {
+                        shardsFromFilters(filters, options)
+                      } else {
+                        shardMapperFunc.assignedShards
```

Comment:
I wonder if this should be in FiloDB itself or an external service (just pass along all of the shards there as an override).